htdocs/core/class/notify.class.php

```diff
@@ -266,6 +269,20 @@ function send($notifcode, $object)
         else $sql.= " AND a.code = '".$notifcode."'";	// New usage
         $sql .= " AND s.rowid = ".$object->socid;
 
+		// Check notification per user
+        $sql.= "\nUNION\n";
+		$sql.= "SELECT  1 as user, c.email, c.rowid as cid, c.lastname, c.firstname, '$langs->defaultlang' as default_lang,";
+		$sql.= " a.rowid as adid, a.label, a.code, n.rowid, n.type";
+        $sql.= " FROM ".MAIN_DB_PREFIX."user as c,";
+        $sql.= " ".MAIN_DB_PREFIX."c_action_trigger as a,";
+        $sql.= " ".MAIN_DB_PREFIX."notify_def as n,";
+        $sql.= " ".MAIN_DB_PREFIX."element_contact as ec";
+        $sql.= " WHERE n.fk_user = c.rowid AND a.rowid = n.fk_action";
+        $sql.= " AND n.fk_user = ec.fk_socpeople";
```

Comment:
fk_socpeople can contains both id of a contact (llx_socpeople) and id of user (llx_user).
If you make this link, you may report notification to user that does not ask it just because a contact was set for it.
For notification "per user", you must use the field "fk_user" of table notify_def like we used the field fk_soc for notification to thirdparties.